docs/key-issues/multiple_architectures.md

```diff
@@ -0,0 +1,259 @@
+# Platforms with multiple CPU architectures
+
+In addition to any ABI requirements, a binary is compiled for a CPU
+architecture. That CPU architecture defines the CPU instructions that can be
+issued by the binary.
+
+## Current state
+
+Historically, it could be assumed that an executable or library would be
+compiled for a single CPU archicture. On the rare occasion that an operating
+system was available for mulitple CPU architectures, it became the
+responsibility of the user to find (or compile) a binary that was compiled for
+their host CPU architecture.
+
+However, we now see operating system platforms where multiple CPU architectures
+are supported:
+
+* In the early days of Windows NT, both x86 and DEC Alpha CPUs were supported
+* Windows 10 supports x86, x86-64, ARMv7 and ARM64; Windows 11 supports x86-64
+  and ARM64.
+* Although Linux started as an x86 project, the Linux kernel is now available a
+  wide range of other CPU architectures, including ARM64, RISC-V, PowerPC, s390
+  and more.
+* Apple transitioned Mac hardware from PowerPC to Intel (x86-64) CPUs, providing
+  a forwards compatibility path for binaries
+* Apple is currently transitioning Mac hardware from Intel (x86-64) to
+  Apple Silicon (ARM64) CPUs, again providing a forwards compatibility
+  path
+* Apple supports ARMv6, ARMv7, ARMv7s, ARM64 and ARM64e on iOS
+* Android currently supports ARMv7, ARM64, x86, and x86-64; it has historically
+  also supported ARMv5 and MIPS
+
+CPU architecture compatibility is a necessary, but not sufficient criterion for
+determining binary compatibility. Even if two binaries are compiled for the same
+CPU architecture, that doesn't guarantee [ABI compatibility](abi.md).
+
+In some respects, CPU architecture compatibility could be considered a superset
+of [GPU compatibility](gpus.md). When dealing with multiple CPU architectures,
+there may be some overal with the solutions that can be used to support GPUs in
+native binaries.
```

Comment:
typo
```suggestion
In some respects, CPU architecture compatibility could be considered a superset
of [GPU compatibility](gpus.md). When dealing with multiple CPU architectures,
there may be some overlap with the solutions that can be used to support GPUs in
native binaries.
```

I also don't see how you mean "superset". GPU architectures are fewer and more homogeneous than CPU architectures. Or do you mean how CUDA is encouraging fat binaries for compatibility (and avoiding the JIT-overhead if it's not pre-compiled)?

Reply:
I mean a "superset" in the sense of that it's all the same problems/restrictions that exist for CPU compatibility, plus some more.